### ISSUE TITLE
Support static props

### ISSUE THREAD

**Issue by aminya** [solidjs-community/eslint-plugin-solid#78]
**Describe the need**

`props` in Solid components can be both reactive and also static. The current rules, especially, `solid-js/reactivity` gives warnings on the static usage of props. When I assign props to a constant or destructure it, I meant to make the property constant. However, the plugin detects that as an error. 

**Suggested Solution**

I would like to see a `static` prefix to be added to the list of ignored props for this warning. The following should be ignored because of the `static` prefix.

```js
const staticProp = props.staticProp
```

**Possible Alternatives**

There's no way to indicate this other than using `eslint-disable` or using `initial/default` prefixes. Both are undesirable for my intention.

**Additional context**

- [x] I would be willing to contribute a PR to implement this feature

**Comment by joshwilsonvu** [comment]
Yes, I'd welcome a change to make `solid/reactivity` accept `static` as a prefix wherever `initial`and `default` are used. I think that's a good solution as `static` would be visible to the caller to help prevent accidentally passing a reactive value to these props. 

In the future I could envision the rule actually warning on `staticProp={signal()}` in the call site. I assume you have no problem with that? But let's just do what you suggested for now.

**Comment by emdede** [comment]
Would like to chime in here.
This is a very useful and far-reaching enhancement. It will shape the solid DX in that it makes you re-evaluate which props need to be dynamically tracked and which are not. As such, I think it would be great to have a more concise prefix that is a bit easier to write than `static` and by having a symbol-based prefix also leaves the realm of regular semantics. I'm talking something along the lines of `!prop` or `_prop` or `$prop`. None of these are well suited, but I hope you get the idea. Denoting something as having a distinct feature, while staying in the realm of regular semantics - by using a literal denomination, i.e. `static` - is not ideal imo.

That said: +1, this is needed and I'm looking forward to its release!
